src/services/requestSessions.js

```diff
@@ -0,0 +1,564 @@
+					user_details: friendDetailsMap[requestSession.friend_id] || null,
+				}
+			})
+
+			const userIds = requestSessionWithDetails.map((item) => item.friend_id)
+			const userDetails = await userRequests.getListOfUserDetails(userIds, true)
+			const userDetailsMap = new Map(userDetails.result.map((userDetail) => [String(userDetail.id), userDetail]))
+			requestSessionWithDetails = requestSessionWithDetails.filter((requestSessionWithDetail) => {
+				const user_id = String(requestSessionWithDetail.friend_id)
+
+				if (userDetailsMap.has(user_id)) {
+					const userDetail = userDetailsMap.get(user_id)
+					requestSessionWithDetail.user_details.image = userDetail.image
+					return true
+				}
+				return false
+			})
+
+			return responses.successResponse({
+				statusCode: httpStatusCode.ok,
+				message: 'SESSION_REQUESTS_LIST',
+				result: { data: requestSessionWithDetails, count: allRequestSession.count },
+			})
+		} catch (error) {
+			console.error(error)
+			throw error
+		}
+	}
+
+	/**
+	 * Accept a pending session request.
+	 * @param {Object} bodyData - The body data containing the target user ID.
+	 * @param {string} bodyData.user_id - The ID of the target user.
+	 * @param {string} mentorUserId - The ID of the authenticated user.
+	 * @param {string} organization_id - the ID of the user organization.
+	 * @returns {Promise<Object>} A success response indicating the request was accepted.
+	 */
+	static async accept(bodyData, mentorUserId, orgId, isMentor, notifyUser) {
+		try {
+			const skipValidation = true
+			const getRequestSessionDetails = await sessionRequestQueries.findOneRequest(
+				mentorUserId,
+				bodyData.user_id,
+				bodyData.start_date,
+				bodyData.end_date
+			)
+			Object.assign(bodyData, {
+				type: common.SESSION_TYPE.PRIVATE,
+				mentor_id: mentorUserId,
+				mentees: [bodyData.user_id],
+				description: getRequestSessionDetails.agenda,
+				title: getRequestSessionDetails.title,
+				start_date: getRequestSessionDetails.start_date,
+				end_date: getRequestSessionDetails.end_date,
+				meta: getRequestSessionDetails.meta ? getRequestSessionDetails.meta : null,
+			})
+			const sessionCreation = await sessionService.create(
+				bodyData,
+				mentorUserId,
+				orgId,
+				isMentor,
+				notifyUser,
+				skipValidation
+			)
+
+			// If the session creation fails, return the actual error message
+			if (sessionCreation.statusCode != httpStatusCode.created) {
+				return responses.failureResponse({
+					statusCode: sessionCreation.statusCode || httpStatusCode.bad_request,
+					message: sessionCreation.message || 'SESSION_CREATION_FAILED',
+					data: sessionCreation.data || [],
+				})
+			}
+
+			const connectionExists = await connectionQueries.getConnection(mentorUserId, bodyData.user_id)
+
+			const approveSessionRequest = await sessionRequestQueries.approveRequest(
+				mentorUserId,
+				bodyData.user_id,
+				getRequestSessionDetails.start_date,
+				getRequestSessionDetails.end_date,
```

Comment:
@sumanvpacewisdom why you are passing start_date and end_date to accept method ?